pkg/handler3/handler_test.go

```diff
@@ -141,49 +147,82 @@ func TestRegisterOpenAPIVersionedService(t *testing.T) {
 			urlPath:      "openapi/v3/apis/apps/v1",
 			respBody:     []byte{},
 		}, {
-			acceptHeader: "application/test,  */*",
-			respStatus:   200,
-			urlPath:      "openapi/v3/apis/apps/v1",
-			respBody:     returnedJSON,
-			expectedETag: computeETag(returnedJSON),
+			acceptHeader:              "application/test,  */*",
+			respStatus:                200,
+			urlPath:                   "openapi/v3/apis/apps/v1",
+			respBody:                  returnedJSON,
+			expectedETag:              computeETag(returnedJSON),
+			responseContentTypeHeader: "application/json",
 		}, {
-			acceptHeader: "application/com.github.proto-openapi.spec.v3@v1.0+protobuf",
-			respStatus:   200,
-			urlPath:      "openapi/v3/apis/apps/v1",
-			respBody:     returnedPb,
-			expectedETag: computeETag(returnedJSON),
+			acceptHeader:              "application/com.github.proto-openapi.spec.v3.v1.0+protobuf",
+			respStatus:                200,
+			urlPath:                   "openapi/v3/apis/apps/v1",
+			respBody:                  returnedPb,
+			expectedETag:              computeETag(returnedJSON),
+			responseContentTypeHeader: "application/com.github.proto-openapi.spec.v3.v1.0+protobuf",
 		}, {
-			acceptHeader: "application/com.github.proto-openapi.spec.v3@v1.0+protobuf",
+			acceptHeader: "application/com.github.proto-openapi.spec.v3.v1.0+protobuf",
 			respStatus:   304,
 			urlPath:      "openapi/v3/apis/apps/v1",
 			respBody:     returnedPb,
 			expectedETag: computeETag(returnedJSON),
 			sendETag:     true,
 		}, {
-			acceptHeader: "application/json, application/com.github.proto-openapi.spec.v2@v1.0+protobuf",
-			respStatus:   200,
-			urlPath:      "openapi/v3/apis/apps/v1",
-			respBody:     returnedJSON,
-			expectedETag: computeETag(returnedJSON),
+			acceptHeader:              "application/json, application/com.github.proto-openapi.spec.v2.v1.0+protobuf",
+			respStatus:                200,
+			urlPath:                   "openapi/v3/apis/apps/v1",
+			respBody:                  returnedJSON,
+			expectedETag:              computeETag(returnedJSON),
+			responseContentTypeHeader: "application/json",
+		}, {
+			acceptHeader:              "application/com.github.proto-openapi.spec.v3.v1.0+protobuf, application/json",
+			respStatus:                200,
+			urlPath:                   "openapi/v3/apis/apps/v1",
+			respBody:                  returnedPb,
+			expectedETag:              computeETag(returnedJSON),
+			responseContentTypeHeader: "application/com.github.proto-openapi.spec.v3.v1.0+protobuf",
 		}, {
-			acceptHeader: "application/com.github.proto-openapi.spec.v3@v1.0+protobuf, application/json",
-			respStatus:   200,
+			acceptHeader: "application/com.github.proto-openapi.spec.v3.v1.0+protobuf, application/json",
+			respStatus:   304,
 			urlPath:      "openapi/v3/apis/apps/v1",
 			respBody:     returnedPb,
 			expectedETag: computeETag(returnedJSON),
+			sendETag:     true,
```

Comment:
We're not setting the `responseContentTypeHeader` here, is it because it's 304? is that the right thing to do @Jefftree ?

Reply:
Per [RFC7323 Section 4.1 304 Not Modified](https://www.rfc-editor.org/rfc/rfc7232#section-4.1) 

>    The server generating a 304 response MUST generate any of the
   following header fields that would have been sent in a 200 (OK)
   response to the same request: Cache-Control, Content-Location, Date,
   ETag, Expires, and Vary.

> Since the goal of a 304 response is to minimize information transfer
   when the recipient already has one or more cached representations, **a
   sender SHOULD NOT generate representation metadata other than the
   above listed fields unless said metadata exists for the purpose of
   guiding cache updates** (e.g., Last-Modified might be useful if the
   response does not have an ETag field).

I think it makes sense to omit the content type here since the the 304 should imply that the body will not be sent and there will be no content type.

Reply:
Nice catch!